lua/lint/linters/zsh.lua

```diff
@@ -1,10 +1,21 @@
+local stdin = not vim.loop.os_uname().version:match("Windows")
+
+local args = {
+  "--no-exec",
+  "--no-rcs",
+  "--no-globalrcs",
+}
+if stdin then
+  table.insert(args, "/dev/stdin")
+end
+
 return {
   cmd = "zsh",
-  stdin = false,
+  stdin = stdin,
   ignore_exitcode = true,
-  args = { "--no-exec" },
+  args = args,
   stream = "stderr",
-  parser = require("lint.parser").from_errorformat("%f:%l:%m", {
+  parser = require("lint.parser").from_errorformat("%s:%l:%m", {
```

Comment:
The matched filename will be meaningless and wrong (`/dev/stdin`) so we can just eat it with `%s`. Without this, the errors don't appear.